api/src/test/java/com/epam/pipeline/manager/pipeline/RunScheduleManagerTest.java

```
@@ -103,179 +121,242 @@ public void setUp() {
+        verify(mockScheduleProviderManager, times(2)).getProvider(ScheduleType.RUN_CONFIGURATION);
     }
 
+
     @Test(expected = IllegalArgumentException.class)
-    @WithMockUser(username = USER_OWNER)
     public void testCreateRunScheduleWithExistentCronExpression() {
+        when(mockScheduleProviderManager.getProvider(ScheduleType.PIPELINE_RUN)).thenReturn(mockPipelineRunScheduleProvider);
+        when(mockAuthManager.getCurrentUser()).thenReturn(mockPipelineUser);
+
         runScheduleManager.createSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
                 Collections.singletonList(testRunScheduleVO));
         testRunScheduleVO.setAction(RunScheduledAction.RESUME);
+
+        doReturn(Collections.singletonList(createRunSchedule())).
+                when(mockRunScheduleDao).loadAllRunSchedulesBySchedulableIdAndType(RUN_ID, ScheduleType.PIPELINE_RUN);
+
         runScheduleManager.createSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
                 Collections.singletonList(testRunScheduleVO));
     }
 
     @Test(expected = IllegalArgumentException.class)
-    @WithMockUser(username = USER_OWNER)
-    public void  createSchedulesWithIdenticalExpressions() {
+    public void createSchedulesWithIdenticalExpressions() {
         final PipelineRunScheduleVO runScheduleVO = getRunScheduleVO(RunScheduledAction.PAUSE, CRON_EXPRESSION1);
         final PipelineRunScheduleVO runScheduleVO2 = getRunScheduleVO(RunScheduledAction.RESUME, CRON_EXPRESSION1);
         runScheduleManager.createSchedules(RUN_ID, ScheduleType.PIPELINE_RUN,
                 Arrays.asList(runScheduleVO, runScheduleVO2));
     }
 
     @Test(expected = IllegalArgumentException.class)
-    @WithMockUser(username = USER_OWNER)
-    public void  createSchedulesWithWrongAction() {
+    public void createSchedulesWithWrongAction() {
+        doThrow(IllegalArgumentException.class).when(mockScheduleProviderManager).getProvider(ScheduleType.RUN_CONFIGURATION);
```

The test is a bit confusing because it is not clear how does a wrong action and `getProvider` throwing `IllegalArgumentException` are connected. I think that `mockScheduleProviderManager.verifyScheduleAction` should throw the exception instead.